packages/dom/src/document.rs

```diff
@@ -243,6 +265,71 @@ impl Document {
             .or(self.try_root_element().map(|el| el.id))
     }
 
+    /// Find the label's bound input elements:
+    /// the element id referenced by the "for" attribute of a given label element
+    /// or the first input element which is nested in the label
+    /// Note that although there should only be one bound element,
+    /// we return all possibilities instead of just the first
+    /// in order to allow the caller to decide which one is correct
+    pub fn label_bound_input_elements(&self, label_node_id: usize) -> Vec<&Node> {
+        let label_node = self.get_node(label_node_id).unwrap();
+        let label_element = label_node.element_data().unwrap();
+        if let Some(target_element_dom_id) = label_element.attr(local_name!("for")) {
+            self.tree()
+                .into_iter()
+                .filter_map(|(_id, node)| {
+                    let element_data = node.element_data()?;
+                    if element_data.name.local != local_name!("input") {
+                        return None;
+                    }
+                    let id = element_data.id.as_ref()?;
+                    if *id == *target_element_dom_id {
+                        Some(node)
+                    } else {
+                        None
+                    }
+                })
+                .collect()
+        } else {
+            label_node
+                .children
+                .iter()
+                .filter_map(|child_id| {
+                    let node = self.get_node(*child_id)?;
+                    let element_data = node.element_data()?;
+                    if element_data.name.local == local_name!("input") {
+                        Some(node)
+                    } else {
+                        None
+                    }
+                })
+                .collect()
+        }
+    }
+
+    pub fn toggle_checkbox(el: &mut ElementNodeData) {
+        let checked_attr_opt = el
+            .attrs
+            .iter_mut()
+            .find(|attr| attr.name.local == local_name!("checked"));
+
+        let checked_attr = if let Some(attr) = checked_attr_opt {
+            attr
+        } else {
+            let attr = Attribute {
+                name: QualName::new(None, ns!(html), local_name!("checked")),
+                value: String::from("false"),
+            };
+            el.attrs.push(attr);
+            el.attrs
+                .iter_mut()
+                .find(|attr| attr.name.local == local_name!("checked"))
+                .unwrap()
+        };
+        let checked = checked_attr.value.parse().unwrap_or(false);
+        checked_attr.value = (!checked).to_string();
+    }
```

Comment:
This needs to actually remove the attribute entirely when unchecking and add it when checking. I suggest using the  empty string for the value to avoid an allocation.